engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/gateway/InclusiveGatewayTest.java

```diff
@@ -574,6 +562,104 @@ public void shouldCompleteScopeOnInclusiveGateway() {
             tuple(processId, ProcessInstanceIntent.COMPLETE_ELEMENT));
   }
 
+  @Test
+  public void shouldCreateDeploymentInclusiveGatewayWithDefaultFlow() {
+    // given
+    final String processId = Strings.newRandomValidBpmnId();
+    // when
+    final BpmnModelInstance processDefinition1 =
+        Bpmn.createExecutableProcess(processId)
+            .startEvent()
+            .inclusiveGateway("inclusive")
+            .sequenceFlowId("s1")
+            .conditionExpression("= contains(str,\"a\")")
+            .endEvent("end1")
+            .moveToLastInclusiveGateway()
+            .defaultFlow()
+            .sequenceFlowId("s2")
+            .endEvent("end2")
+            .done();
+
+    final Record<DeploymentRecordValue> deployment1 =
+        ENGINE.deployment().withXmlResource(processDefinition1).deploy();
+
+    // then
+    assertThat(deployment1.getKey())
+        .describedAs("Inclusive gateway's default flow should be allowed to have no condition")
+        .isNotNegative();
+
+    // when
+    final BpmnModelInstance processDefinition2 =
+        Bpmn.createExecutableProcess(processId)
+            .startEvent()
+            .inclusiveGateway("inclusive")
+            .sequenceFlowId("s1")
+            .conditionExpression("= contains(str,\"a\")")
+            .endEvent("end1")
+            .moveToLastInclusiveGateway()
+            .defaultFlow()
+            .sequenceFlowId("s2")
+            .conditionExpression("= contains(str,\"b\")")
+            .endEvent("end2")
+            .done();
+
+    final Record<DeploymentRecordValue> deployment2 =
+        ENGINE.deployment().withXmlResource(processDefinition2).deploy();
+
+    // then
+    assertThat(deployment2.getKey())
+        .describedAs(
+            "Inclusive gateway's default flow should be allowed to have a condition, but not be required")
+        .isNotNegative();
+  }
+
+  @Test
+  public void shouldTakeDefaultFlowDoesNotRaiseIncident() {
```

Comment:
🔧 The intent of this test is to verify that the default flow's condition is not evaluated. Let's capture that in the test case name.

```suggestion
  public void shouldNotEvaluateConditionOfDefaultFlow() {
```